### PR TITLE
fix!: ConsumerModificationProposal is needed in Gaia upgrade handler

### PR DESCRIPTION
**Pull request by mpoke**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [ ] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] Added `!` to the type prefix if the change is [state-machine breaking](https://github.com/cosmos/interchain-security/blob/main/RELEASES.md#breaking-changes)
* [ ] Confirmed this PR does not introduce changes requiring state migrations, OR migration code has been added to consumer and/or provider modules
* [ ] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] Provided a link to the relevant issue or specification
* [ ] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/00-intro.md)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#testing)
* [ ] Added a changelog entry to `CHANGELOG.md`
* [ ] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [ ] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed
* [ ] If this PR is library API breaking, bump the go.mod version string of the repo, and follow through on a new major release

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` the type prefix if the change is state-machine breaking
* [ ] confirmed this PR does not introduce changes requiring state migrations, OR confirmed migration code has been added to consumer and/or provider modules
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

## Summary by CodeRabbit

- **New Features**
	- Introduced a new proposal type for consumer modifications, enhancing the proposal handling capabilities.
	- Added methods for constructing and validating consumer modification proposals.

- **Improvements**
	- Updated string constants for clarity in consumer proposal types, improving semantic understanding. 

- **Bug Fixes**
	- Implemented validation checks for consumer modification proposals to ensure adherence to criteria before processing.

### CODE REVIEW

File: x/ccv/provider/types/legacy_proposal.go [hunk 1]
```diff
@@ -20,15 +20,16 @@ import (
 
 const (
 	ProposalTypeConsumerAddition     = "ConsumerAddition"
-	ProposalTypeConsumerRemoval      = "RemoveConsumer"
-	ProposalTypeConsumerModification = "UpdateConsumer"
+	ProposalTypeConsumerRemoval      = "ConsumerRemoval"
```
Comment (MSalopek) [inline x/ccv/provider/types/legacy_proposal.go:23]: Why the rename?

File: x/ccv/provider/types/legacy_proposal.go [hunk 2]
```diff
@@ -239,6 +240,55 @@ func (sccp *ConsumerRemovalProposal) ValidateBasic() error {
 	return nil
 }
 
+// NewConsumerModificationProposal creates a new consumer modification proposal.
+func NewConsumerModificationProposal(title, description, chainID string,
```
Comment (MSalopek) [inline x/ccv/provider/types/legacy_proposal.go:244]: Does this need a `[Deprecated]` comment?

**Review by MSalopek — APPROVED** [review]

File: x/ccv/provider/types/legacy_proposal.go (hunk 1 above)
Comment (mpoke) [reply x/ccv/provider/types/legacy_proposal.go:23]: I just reverted the entire file to what's in ICS main.

File: x/ccv/provider/types/legacy_proposal.go (hunk 2 above)
Comment (mpoke) [reply x/ccv/provider/types/legacy_proposal.go:244]: @bermuell ?

**Review by insumity — APPROVED** [review]

File: x/ccv/provider/types/legacy_proposal.go (hunk 2 above)
Comment (insumity) [reply x/ccv/provider/types/legacy_proposal.go:244]: I would add a `[Deprecated]` comment. It should be deprecated after we get Permissionless.

**Pull request merged**